lms/lib/comment_client/comment.py

```diff
@@ -27,9 +27,15 @@ class Comment(models.Model):
     base_url = "{prefix}/comments".format(prefix=settings.PREFIX)
     type = 'comment'
 
+    def __init__(self, *args, **kwargs):
+        super(Comment, self).__init__(*args, **kwargs)
+        self.cached_thread = None
+
     @property
     def thread(self):
-        return Thread(id=self.thread_id, type='thread')
+        if not self.cached_thread:
```

Comment:
This assumes there are no false values for Thread, which seems likely to be a valid assumption.